src/lib/analyses.ts

```diff
@@ -32,54 +150,89 @@ export enum AggregateRecommendationDecision {
 }
 
 export interface AggregateRecommendation {
+  analysisStrategy: AnalysisStrategy
   decision: AggregateRecommendationDecision
   chosenVariationId?: number
+  statisticallySignificant?: boolean
+  practicallySignificant?: PracticalSignificanceStatus
+}
+
+const PracticalSignificanceStatusToDecision: Record<PracticalSignificanceStatus, AggregateRecommendationDecision> = {
+  [PracticalSignificanceStatus.No]: AggregateRecommendationDecision.DeployAnyVariation,
+  [PracticalSignificanceStatus.Uncertain]: AggregateRecommendationDecision.Inconclusive,
+  [PracticalSignificanceStatus.Yes]: AggregateRecommendationDecision.DeployChosenVariation,
 }
 
 /**
- * Returns the aggregate recommendation over analyses of different analysis strategies.
+ * Returns the recommendation for a single analysis.
  *
- * @param analyses Analyses of different strategies for the same day.
- * @param defaultStrategy Default strategy in the context of an aggregateRecommendation..
+ * See file-level recommendation documentation.
  */
-export function getAggregateRecommendation(
-  analyses: Analysis[],
-  defaultStrategy: AnalysisStrategy,
+export function getMetricAssignmentRecommendation(
+  experiment: ExperimentFull,
+  metric: MetricBare,
+  analysis: Analysis,
 ): AggregateRecommendation {
-  const recommendationChosenVariationIds = analyses
-    .map((analysis) => analysis.recommendation)
-    .map((recommendation) => recommendation?.chosenVariationId)
-    .filter((x) => x)
-  const recommendationConflict = [...new Set(recommendationChosenVariationIds)].length > 1
-  if (recommendationConflict) {
+  const metricAssignment = experiment.metricAssignments.find(
+    (metricAssignment) => metricAssignment.metricAssignmentId === analysis.metricAssignmentId,
+  )
+  const diffCredibleIntervalStats =
+    analysis && metricAssignment && getDiffCredibleIntervalStats(analysis, metricAssignment)
```

Comment:
There shouldn't be a need to check the `analysis` again, and we should be able to assume that the metric assignment exists.

Also, `getDiffCredibleIntervalStats()` can handle a null `analysis` and line 177 assumes a non-null/undefined `analysis`.

```suggestion
  const metricAssignment = experiment.metricAssignments.find(
    (metricAssignment) => metricAssignment.metricAssignmentId === analysis.metricAssignmentId,
  ) as MetricAssignment
  const diffCredibleIntervalStats = getDiffCredibleIntervalStats(analysis, metricAssignment)
```

Reply:
I will remove the analysis, but there I think it is worth keeping the metricAssignment check - there have been backend/etl bugs before causing the front-end to crash.